### DISCUSSION TITLE
How to learn web assembly in order to create a language on top of it.

### DISCUSSION THREAD

**Discussion by Huzaifa-MS** [WebAssembly/design#1496] [General]
What steps can you recommend I take to learn web assembly to have all its features in a programming language I want to make on top of it? For example how could I study Go and see how it compiles to wasm?

**Comment by fgmccabe** [comment]
I would not recommend using go-lang for this: the mapping to WebAssembly for go-lang is 'tortuous'.
Easier just to read the spec.

**Comment by fgmccabe** [comment]
Another choice is to use godbolt.org; there is an option to generate webassembly from C there.

**Reply by Huzaifa-MS** [reply]
Thank you for replying. I appreciate it.
